packages/@glimmer/component/src/-private/ember-component-manager.ts

```diff
@@ -0,0 +1,49 @@
+import { destroy } from '@ember/destroyable';
+import { capabilities } from '@ember/component';
+import { schedule } from '@ember/runloop';
+import BaseComponentManager from './base-component-manager';
+
+import { type default as GlimmerComponent, setDestroyed, setDestroying } from './component';
+import type { Arguments } from '@glimmer/interfaces';
+
+const CAPABILITIES = capabilities('3.13', {
+  destructor: true,
+  asyncLifecycleCallbacks: false,
+  updateHook: false,
+});
+
+function scheduledDestroyComponent(component: GlimmerComponent): void {
+  if (component.isDestroyed) {
+    return;
+  }
+
+  destroy(component);
+  setDestroyed(component);
+}
+
+/**
+ * This component manager runs in Ember.js environments and extends the base component manager to:
+ *
+ * 1. Properly destroy the component's associated `meta` data structure
+ * 2. Schedule destruction using Ember's runloop
+ */
+class EmberGlimmerComponentManager extends BaseComponentManager<GlimmerComponent> {
+  capabilities = CAPABILITIES;
+
+  destroyComponent(component: GlimmerComponent): void {
+    if (component.isDestroying) {
+      return;
+    }
+
+    setDestroying(component);
+
+    schedule('actions', component, component.willDestroy);
```

Comment:
where is this type of destruction scheduled in specific queues? does all destruction work like this?
I don't know that I've seen this outside of `@glimmer/component`

`destroy` in `@glimmer/destroyable` does have scheduling behavior as well :thinking:  https://github.com/glimmerjs/glimmer-vm/blob/main/packages/%40glimmer/destroyable/index.ts#L153 -- but it's different.